Terminal.Gui/View/Layout/Dim.cs

```diff
@@ -0,0 +1,766 @@
+    /// </summary>
+    /// <returns></returns>
+    internal virtual bool ReferencesOtherViews () { return false; }
+
+    /// <inheritdoc/>
+    public override bool Equals (object other) { return other is Dim abs && abs == this; }
+
+    /// <inheritdoc/>
+    public override int GetHashCode () { return Anchor (0).GetHashCode (); }
+
+    #endregion overrides
+}
+
+/// <summary>
+///     Represents a dimension that is a fixed size.
+/// </summary>
+/// <remarks>
+///     <para>
+///         This is a low-level API that is typically used internally by the layout system. Use the various static
+///         methods on the <see cref="Dim"/> class to create <see cref="Dim"/> objects instead.
+///     </para>
+/// </remarks>
+/// <param name="size"></param>
+public class DimAbsolute (int size) : Dim
+{
+    /// <inheritdoc/>
+    public override bool Equals (object other) { return other is DimAbsolute abs && abs.Size == Size; }
+
+    /// <inheritdoc/>
+    public override int GetHashCode () { return Size.GetHashCode (); }
+
+    /// <summary>
+    ///     Gets the size of the dimension.
+    /// </summary>
+    public int Size { get; } = size;
+
+    /// <inheritdoc/>
+    public override string ToString () { return $"Absolute({Size})"; }
+
+    internal override int Anchor (int size) { return Size; }
+
+    internal override int Calculate (int location, int superviewContentSize, View us, Dimension dimension)
+    {
+        // DimAbsolute.Anchor (int size) ignores width and returns n
+        return Math.Max (Anchor (0), 0);
+    }
+}
+
+/// <summary>
+///     Represents a dimension that automatically sizes the view to fit all the view's Content, SubViews, and/or Text.
+/// </summary>
+/// <remarks>
+///     <para>
+///         See <see cref="DimAutoStyle"/>.
+///     </para>
+///     <para>
+///         This is a low-level API that is typically used internally by the layout system. Use the various static
+///         methods on the <see cref="Dim"/> class to create <see cref="Dim"/> objects instead.
+///     </para>
+/// </remarks>
+/// <param name="style">
+///     Specifies how <see cref="DimAuto"/> will compute the dimension. The default is <see cref="DimAutoStyle.Auto"/>.
+/// </param>
+/// <param name="minimumContentDim">The minimum dimension the View's ContentSize will be constrained to.</param>
+/// <param name="maximumContentDim">The maximum dimension the View's ContentSize will be fit to. NOT CURRENTLY SUPPORTED.</param>
+public class DimAuto (DimAutoStyle style, Dim minimumContentDim, Dim maximumContentDim) : Dim
+{
+    /// <inheritdoc/>
+    public override bool Equals (object other)
+    {
+        return other is DimAuto auto && auto.MinimumContentDim == MinimumContentDim && auto.MaximumContentDim == MaximumContentDim && auto.Style == Style;
+    }
+
+    /// <inheritdoc/>
+    public override int GetHashCode () { return HashCode.Combine (base.GetHashCode (), MinimumContentDim, MaximumContentDim, Style); }
```

Comment:
Love the use of Combine. That's disappointingly rare in the wild, even though it's the easiest, recommended, standard, and generally best way to do this.

Otherwise, same comments as above, of course.